src/jsonpatcherproxy.js

```diff
@@ -143,7 +144,9 @@ var JSONPatcherProxy = (function() {
         return Reflect.deleteProperty(target, key);
       }
     });
-    return proxy;
+    /* keeping track of all the proxies to be able to revoke them later */
+    this.proxifiedObjectsMap.push(proxy);
+    return proxy.proxy;
   };
   //grab tree's leaves one by one, encapsulate them into a proxy and return
   JSONPatcherProxy.prototype._proxifyObjectTreeRecursively = function(
```

Comment:
Could you use jsfiddle.net?

Once delete patch is applied binding should be removed as well.
If you are talking about race conditions between browser-to-server and server-to-browser messages, OT should solve it.